### PR TITLE
DashboardScene: Backward compatability with getDashboardSrv.getCurrent and DashboardModel props and functions

### PR DESCRIPTION
**Pull request by torkelo**
Just a start to provide some basic level of compatability, something we need to expand on and fill in missing holes as we discover them

### CODE REVIEW

File: public/app/plugins/panel/alertlist/UnifiedAlertList.tsx [hunk 1]
```diff
@@ -159,7 +159,9 @@ export function UnifiedAlertList(props: PanelProps<UnifiedAlertListOptions>) {
     if (!promRulesRequests.loading) {
       fetchPromAndRuler({ dispatch, limitInstances, matcherList, dataSourceName, stateList });
     }
+
     const sub = dashboard?.events.subscribe(TimeRangeUpdatedEvent, () => {
+      console.log('TimeRangeUpdatedEvent');
```
Comment (dprokop) [inline public/app/plugins/panel/alertlist/UnifiedAlertList.tsx:164]:
```suggestion
```

File: public/app/features/dashboard-scene/utils/DashboardModelCompatibilityWrapper.ts [hunk 2]
```diff
@@ -1,17 +1,67 @@
...
+  public get time() {
+    const time = sceneGraph.getTimeRange(this._scene);
+    return {
+      from: time.state.from,
+      to: time.state.to,
+    };
+  }
+
+  /**
+   * Used from from timeseries migration handler to migrate time regions to dashboard annotations
```
Comment (dprokop) [inline public/app/features/dashboard-scene/utils/DashboardModelCompatibilityWrapper.ts:56]: Do you mean this migration https://github.com/grafana/grafana/blob/main/public/app/plugins/panel/timeseries/migrations.ts#L65 ? Ah it's gonna be interesting to implement as the handler is mutating the original dashboard. Doable of course via setter, but gonna be funky.

File: public/app/plugins/panel/alertlist/UnifiedAlertList.tsx (hunk 1 above)
Comment (dprokop) [reply public/app/plugins/panel/alertlist/UnifiedAlertList.tsx:164]: Actually changes in this entire file are to be reverted :)

**Review by dprokop — APPROVED** [review]
This is great,  I'll turn the TODOs from this file into a followup issue to get this done independently from when we find the gaps.

File: public/app/features/dashboard-scene/utils/DashboardModelCompatibilityWrapper.ts (hunk 2 above)
Comment (dprokop) [reply public/app/features/dashboard-scene/utils/DashboardModelCompatibilityWrapper.ts:56]: `__grafanaSceneContext` is a clever idea actually, i like it

File: public/app/plugins/panel/alertlist/UnifiedAlertList.tsx (hunk 1 above)
Comment (torkelo) [reply public/app/plugins/panel/alertlist/UnifiedAlertList.tsx]: yea, reverted these changes :)

**Pull request merged**